From 0a3cfd65062d21816d06ad6f71bb0222b09e7605 Mon Sep 17 00:00:00 2001
From: Samuel Williams <samuel.williams@oriontransfer.co.nz>
Date: Wed, 4 Mar 2026 20:53:23 +1300
Subject: [PATCH] Test robustness on Ruby 4+.

--- a/test/async/http/proxy.rb
+++ b/test/async/http/proxy.rb
@@ -161,6 +161,12 @@
 					reader.wait
 					writer.wait
 				ensure
+					reader.stop
+					reader.wait rescue nil
+					
+					writer.stop
+					writer.wait rescue nil
+					
 					upstream.close
 					stream.close
 				end
